build_tools/teatime.py

```diff
@@ -103,6 +110,66 @@ def finish(self):
                 b"[" + self.label + b" completed in " + run_pretty.encode() + b"]\n"
             )
 
+        # Call coordinate_index_and_logs if upload is configured
+        if self.upload_to_s3 and self.s3_bucket and self.s3_subdir:
+            self.coordinate_index_and_logs()
```

Comment:
teatime.py is called once _per subproject_: https://github.com/ROCm/TheRock/blob/25516d0df8821844845670d59a0bf8e18af66390/cmake/therock_subproject.cmake#L97-L109

Is this index page generation and uploading safe to run multiple times? Would it make more sense to generate the index page at the start or end of the build? I think ideally there would be a dynamic index page running on the server that would be aware of which logs have already been uploaded, rather than need to regenerate a static index page 10-20 times.